### PR TITLE
[AAQ-559] remove service identity var and improve rag reasoning

### PR DESCRIPTION
**Pull request by suzinyou**
Reviewer: @lickem22 
Estimate: 30 min

---

## Ticket

Fixes: https://idinsight.atlassian.net/browse/AAQ-559

### Goal

Overall improve the RAG pipeline

### Changes

1. Improve reasoning in RAG prompt and return a json object
2. Remove SERVICE_IDENTITY from prompts -- should we just remove the topic check also?
3. Also removed SA languages we don't support

### Future Tasks (optional)

- [ ] improve groundedness & response language of output

## How has this been tested?

- locally with docker-compose
- on AAQ deployed instance
- unit tests
- Also tested alignscore: 38/48 cases passed

## Checklist

Fill with `x` for completed. 

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts

(Delete any items below that are not relevant)
- [x] I have updated the automated tests
- [ ] I have added a blogpost in Latest Updates  --> I will write one after further improvements this sprint!

### CODE REVIEW

File: core_backend/app/llm_call/llm_prompts.py [hunk 1]
```diff
@@ -206,24 +204,50 @@ def get_prompt(cls) -> str:
...
 
 
-# ----  Question answering bot
-ANSWER_FAILURE_MESSAGE = "Sorry, no relevant information found."
-ANSWER_QUESTION_PROMPT = f"""
-You are a question answering system that is constantly learning and improving.
-You can process and comprehend many pieces of text and utilize this knowledge to \
-provide concise, accurate, and informative answers to diverse queries.
+# ---- Response generation
+ANSWER_FAILURE_MESSAGE = "FAILED"
```
Comment (suzinyou) [inline core_backend/app/llm_call/llm_prompts.py:208]: There's no need for the failure message to be long! So I changed it

File: core_backend/app/llm_call/utils.py [hunk 2]
```diff
@@ -37,10 +42,16 @@ async def _ask_llm_async(
...
+        **extra_kwargs,
     )
     logger.info(f"LLM output: {llm_response_raw.choices[0].message.content}")
-
     return llm_response_raw.choices[0].message.content
+
+
+def remove_json_markdown(text: str) -> str:
+    """Remove json markdown from text."""
+    return text.removeprefix("```json").removesuffix("```").strip()
```
Comment (suzinyou) [inline core_backend/app/llm_call/utils.py:57]: Sometimes LLMs like to format json output as markdowns.

File: core_backend/app/question_answer/utils.py [hunk 3]
```diff
@@ -26,6 +26,10 @@ def get_context_string_from_retrieved_contents(
     """
     context_list = []
     for i, result in content_response.items():
-        context_list.append(f"{i+1}. {result.retrieved_title}\n{result.retrieved_text}")
+        if not isinstance(result, QuerySearchResult):
+            result = QuerySearchResult(**result)
+        context_list.append(
+            f"{int(i)+1}. {result.retrieved_title}\n{result.retrieved_text}"
+        )
```
Comment (suzinyou) [inline core_backend/app/question_answer/utils.py:33]: This is useful for evaluation when I do /embeddings-search first, and use that response to test different RAG functions (prompts/rails)

File: core_backend/app/llm_call/llm_prompts.py (hunk 1 above)
Comment (lickem22) [reply core_backend/app/llm_call/llm_prompts.py:208]: Is this the responsed sent to the user or the response to the paraphrasing. If it is that later maybe "PARAPHRASHING FAILED"would be more intuitive

File: core_backend/app/config.py [hunk 4]
```diff
@@ -24,7 +24,7 @@
 # for all of its endpoints.
 LITELLM_MODEL_EMBEDDING = os.environ.get("LITELLM_MODEL_EMBEDDING", "openai/embeddings")
 LITELLM_MODEL_DEFAULT = os.environ.get("LITELLM_MODEL_DEFAULT", "openai/default")
-LITELLM_MODEL_SUMMARIZATION = os.environ.get(
+LITELLM_MODEL_GENERATION = os.environ.get(
     "LITELLM_MODEL_SUMMARIZATION", "openai/generate-response"
```
Comment (lickem22) [inline core_backend/app/config.py:28]: Should we update the environment variable name as well?

Comment (suzinyou) [reply core_backend/app/config.py:28]: Omg yes.

File: core_backend/app/llm_call/llm_prompts.py (hunk 1 above)
Comment (suzinyou) [reply core_backend/app/llm_call/llm_prompts.py:208]: We should prob clean up the terminology.. this is just for RAQ response generation failure. This is only used internally if response generation failed. @amiraliemami had some ideas to clean up actual error codes depending on where in the guardrails/RAG pipeline they come from

File: core_backend/app/config.py (hunk 4 above)
Comment (suzinyou) [reply core_backend/app/config.py]: Fixed! Thank you for catching this!

**Pull request merged**